.github/workflows/release-cde-job.yml

```diff
@@ -0,0 +1,362 @@
+          service_account: ${{ vars.RUNNER_SERVICE_ACCOUNT }}        
+          docker_enabled: true
+    outputs:
+      runner_label: ${{ steps.create-runner.outputs.runner_label }}
+  login-poetry:
+    name: Login Poetry
+    runs-on: [self-hosted, "${{ needs.create-runner.outputs.runner_label }}"]
+    needs:
+      - create-runner
+    steps:
+      - id : login-poetry
+        run : |
+            echo "POETRY_HTTP_BASIC_ARTIFACT_PASSWORD=$(aws codeartifact get-authorization-token --domain-owner ${{ secrets.AWS_ACCOUNT_ID }} \
+              --domain ${{ vars.CODEARTIFACT_DOMAIN }} --query 'authorizationToken' --output text)" >> "$GITHUB_OUTPUT"
+      - id : login-ecr
+        run : |
+            echo "ECR_PASSWORD=$(aws ecr get-login-password --region eu-central-1)" >> "$GITHUB_OUTPUT"
+    outputs:
+      poetry_token: ${{ steps.login-poetry.outputs.POETRY_HTTP_BASIC_ARTIFACT_PASSWORD }}
+      docker_password: ${{ steps.login-ecr.outputs.ECR_PASSWORD }}
+  build-venv:
+    name: Build Env
+    needs:
+      - create-runner
+      - login-poetry 
+    runs-on: [self-hosted, "${{ needs.create-runner.outputs.runner_label }}"]
+    container:
+      image: ${{ vars.CDE_RUNTIME_IMAGE }}
+      credentials:
+        username: AWS
+        password: ${{ needs.login-poetry.outputs.docker_password }}
+      options: 
+        --privileged
+        --user root
+      env:
+        POETRY_HTTP_BASIC_ARTIFACT_PASSWORD: ${{ needs.login-poetry.outputs.poetry_token }}
+    env:
+      POETRY_HTTP_BASIC_ARTIFACT_PASSWORD: ${{ needs.login-poetry.outputs.poetry_token }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build Project
+        continue-on-error: false
+        working-directory: ${{ inputs.working-dir }}
+        run: |
+          poetry config virtualenvs.in-project true
+          poetry config virtualenvs.create true
+          export POETRY_HTTP_BASIC_ARTIFACT_USERNAME=aws
+          #poetry lock --no-update
+          poetry install --without test,dev --no-root
+          poetry build -f wheel -n 
+          poetry run pip install --no-deps dist/*.whl
+          poetry run pip install venv-pack2
+          poetry run venv-pack -o ${{ inputs.job-name }}.tar.gz
+      - name: Upload Build Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-artifact
+          path: ${{ inputs.working-dir }}/${{ inputs.job-name }}.tar.gz
+          overwrite: true
+  release:
+    name: Release
+    runs-on: [self-hosted, "${{ needs.create-runner.outputs.runner_label }}"]
+    needs:
+      - create-runner
+      - login-poetry
+      - build-venv
+    outputs:
+      version: ${{ steps.release.outputs.tag }}
+    if: needs.build-venv.result == 'success'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Final Semantic Release
+        uses: python-semantic-release/python-semantic-release@v9.8.8
```

Comment:
pin the hash instead of the tag, this comment applies on each external used action